auth/store.go

```diff
@@ -237,8 +253,29 @@ func (as *authStore) Authenticate(ctx context.Context, username, password string
 		return &pb.AuthenticateResponse{}, ErrAuthFailed
 	}
 
-	token := fmt.Sprintf("%s.%d", simpleToken, index)
-	as.assignSimpleTokenToUser(username, token)
+	var token string
+
+	switch as.tokenType {
+	case TokenSimple:
+		simpleToken := ctx.Value("simpleToken").(string)
+		token = fmt.Sprintf("%s.%d", simpleToken, index)
+		as.assignSimpleTokenToUser(username, token)
+	case TokenJwt:
+		t := jwt.NewWithClaims(jwt.GetSigningMethod(as.signMethod),
```

Comment:
it probably worth to have separate funcs to do jwt token generation and verification.